### PR TITLE
SAX buf - avoid overflow

### PR DESCRIPTION
**Pull request by clod81**
check that the actual size of dest buffer before performing the strncpy

reference https://github.com/ohler55/ox/issues/195

### CODE REVIEW

File: ext/ox/sax_buf.c [hunk 1]
```diff
@@ -201,6 +201,9 @@ ox_stpncpy(char *dest, const char *src, size_t n) {
     if (n < cnt) {
 	cnt = n;
     }
+    if (cnt > sizeof(dest) - 1){
```
Comment (ohler55) [inline ext/ox/sax_buf.c]: Size of a pointer will be 8 or 4 depending on the architecture. Not really what we want. The problem will be further up. Probably where n is provided.

File: ext/ox/sax_buf.c [hunk 2]
```diff
@@ -203,7 +203,7 @@ ox_stpncpy(char *dest, const char *src, size_t n) {
 	cnt = n;
     }
     if (cnt > dest_s){
-      cnt = dest_s - 1;
+      cnt = dest_s;
```
Comment (ohler55) [inline ext/ox/sax_buf.c]: Probably was for a trailing `\0` character. I'll have to check first.

File: ext/ox/sax_buf.c (hunk 1 above)
Comment (clod81) [reply ext/ox/sax_buf.c]: true. will delete the PR

**Comment by ohler55** [comment]
It looks like the buffer tails is after the end. That should not happen so I'll look for where that happens.

**Pull request closed without merging**